syntax = "proto3";

package ai_project_helper;

service AIProjectHelper {
  rpc RunPlan(PlanExecuteRequest) returns (stream ActionFeedback);
  rpc GetPlan(PlanGetRequest) returns (stream ActionFeedback); // 新增方法
  rpc GetPlanThenRun(PlanThenExecuteRequest) returns (stream ActionFeedback);
 
}

message PlanExecuteRequest {
  string plan_text = 1;
  string project_id = 2;
}

message PlanGetRequest {
  string requirement = 1;
  string model = 2;
  string llm_url = 3;
  string project_id = 4;
}

message PlanThenExecuteRequest {
  string requirement = 1;
  string model = 2;
  string llm_url = 3;
  string project_id = 4;
}

message ActionFeedback {
  int32 action_index = 1;
  string action_type = 2;
  string step_description = 3;
  string status = 4;
  string output = 5;
  string error = 6;
  string command = 7;
  
  int32 step_index = 8;     // 当前步骤索引
  int32 total_steps = 9;    // 总步骤数
  int32 exit_code = 10;     // 新增退出码字段
  string complete_plan = 11; 
}
